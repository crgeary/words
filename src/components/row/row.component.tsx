import { FC } from "react";
import { VALID_WORD_LENGTH } from "../../constants";

import { Tile, TileProps } from "../tile/tile.component";

export type RowProps = {
    letters: Pick<TileProps, "children" | "state">[];
};

export const Row: FC<RowProps> = ({ letters }) => {
    return (
        <div className="grid grid-cols-5 gap-1">
            {Array.from({ ...letters, length: VALID_WORD_LENGTH }).map((letter, i) => (
                <Tile key={i} {...letter} />
            ))}
        </div>
    );
};
