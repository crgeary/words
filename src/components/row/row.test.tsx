import { render } from "@testing-library/react";
import { Row } from "./row.component";

test("it should match snapshot", () => {
    const letters = "hello";
    const { container } = render(
        <Row letters={letters.split("").map((children) => ({ children }))} />,
    );
    expect(container.firstChild).toMatchSnapshot();
});
