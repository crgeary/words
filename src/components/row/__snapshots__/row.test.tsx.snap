// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot 1`] = `
<div
  class="grid grid-cols-5 gap-1"
>
  <span
    class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
  >
    <span
      class="uppercase font-semibold text-xl"
    >
      h
    </span>
  </span>
  <span
    class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
  >
    <span
      class="uppercase font-semibold text-xl"
    >
      e
    </span>
  </span>
  <span
    class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
  >
    <span
      class="uppercase font-semibold text-xl"
    >
      l
    </span>
  </span>
  <span
    class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
  >
    <span
      class="uppercase font-semibold text-xl"
    >
      l
    </span>
  </span>
  <span
    class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
  >
    <span
      class="uppercase font-semibold text-xl"
    >
      o
    </span>
  </span>
</div>
`;
